### ISSUE TITLE
Tar ball releases

### ISSUE THREAD

**Issue by Imaclean74** [opendistro-for-elasticsearch/community#12]
There is already an issue for alternative distro packages. In addition - tar/zip files are a must. These are java based products meaning that for the most part they will just run - without having specific binary dependencies - so a simple zip file allows the broadest possible platform usage - including windows. And this is the model that elastic themselves use - as well as the other download formats they offer.

**Comment by peterbe** [comment]
My reasoning for this is that I can't use Docker because it's not performant enough. I have a Mac Book Pro and I have a decent dump of the production database here on my laptop. It's about 5GB on disk. At the moment I run it like this:

```
cd /location/of/unzipped/elasticsearch-6.1.0 && ./bin/elasticsearch
```

I just unzipped the .zip I downloaded from [elastic.co](https://www.elastic.co/downloads/elasticsearch). This runs much faster than my Docker for mac. I did try with Docker. I gave it a lot of RAM and a couple of CPUs but it was always just a matter of time till it crashed. Not to mention, it ran slower. 

Perhaps things are different these days (with my now added experience with docker) but numerous times I would hit disk limits with docker and had to wipe those massive Docker for mac files to make it start again. Every time I would lose my local development database and had to start over. 

PS. I use, and like, Docker for many other projects. What I often do is things like setting `DJANGO_ES_URLS=http://docker.for.mac.host.internal:9200` then a bunch of services run inside Docker but they reach back out to the host of the Elasticsearch "cluster". I just have to remember to do some terminal juggling.

**Comment by satyaakam** [comment]
agreed giving a Tar ball will help in testing it out on other platforms than just RPM based ones

**Comment by PACordonnier** [comment]
The tarball will also allow to use the securityadmin.sh from a management machine that does not run an elasticsearch node :)

**Comment by krishnaganjam** [comment]
Is there a way to build tar ball from sources while waiting for this to be available?

**Comment by daveram** [comment]
> Is there a way to build tar ball from sources while waiting for this to be available?

Would also like to know if this is possible, we're currently pulling the rpm apart to deploy as rpm is not an option for us.  Certainly not a clean or proper method, lots of messing around to make it work

**Comment by alolita** [comment]
Hi @krishnaganjam et.al, 
You can get source zips, tarballs for each plug-in from their respective repos - 
Alerting: https://github.com/opendistro-for-elasticsearch/alerting/releases
SQL: https://github.com/opendistro-for-elasticsearch/sql/releases
Security: https://github.com/opendistro-for-elasticsearch/security/releases
Performance Analyzer: https://github.com/opendistro-for-elasticsearch/performance-analyzer/releases

Build scripts for distribution builds are also available at: https://github.com/opendistro-for-elasticsearch/opendistro-build

Hope this helps. Closing this issue.

**Comment by daveram** [comment]
Seems weird that other AWS distributions like Corretto provide tarballs/zips with everything included not requiring folks to go build it themselves, but for Elastic we can't do the same.  Not entirely sure closing this is the right approach

**Comment by poggdogg** [comment]
Agreed, this would be like building a development environment on every production host.  It's quite a bit more overhead than we need.

In secure environments where you can't always run as root this is kinda kludgy.  A portable install option by default would make this less clunky.

**Comment by peterbe** [comment]
What?! Why close this. There is still no tar ball for Elasticsearch. Only for the plugins?

**Comment by poggdogg** [comment]
> What?! Why close this. There is still no tar ball for Elasticsearch. Only for the plugins?

And only _some_ of the plugins, I noticed.

**Comment by poggdogg** [comment]
Followed the process above - the gradle will _**ONLY**_ build packages, not binaries, not tarballs.

**Comment by alolita** [comment]
Hi @poggdogg @peterbe - the source tar balls I listed are for the plugins only. Tarballs for Elasticsearch and Kibana are on the way. We can leave this issue open for tracking the request.

**Comment by Imaclean74** [comment]
thank's @alolita, and you mention those are *source* tar balls. The original request was for zip/tar files of build binaries for the full distro. Is there a plan for that ?   Now that he build scripts are now available - will you be accepting PRs on them ? and are they or will they be running in a publicly viewable CI process ?

**Comment by alolita** [comment]
@Imaclean74 Thanks for clarifying! We are planning to release tarballs for the binaries shortly :-) So please keep a lookout. 
You are most welcome to submit PRs on build scripts. All enhancements, bug fixes, changes are welcome. 
We have on our to-do list to make available a publicly viewable CI process - will share more details on this task as we make progress. Thanks again for checking in!

**Comment by alolita** [comment]
Closing issue. Tarballs were released with 1.2.0. https://discuss.opendistrocommunity.dev/t/open-distro-for-elasticsearch-1-2-0-is-now-available-with-new-linux-tarballs/1565
